services/api-server/src/simcore_service_api_server/models/schemas/solvers.py

```diff
@@ -0,0 +1,170 @@
+    version: str  # complete tag.  e.g. 3.4.5 TODO: regex for version in python PEP
+    version_aliases: List[str] = []  # remaining tags
+
+    title: str
+    description: Optional[str]
+    maintainer: str
+    released: Optional[datetime]  # TODO: turn into required
+
+    solver_url: HttpUrl
+
+    @classmethod
+    def create_from_image(cls, img: SolverImage, tags: List, **kwargs) -> "Solver":
+        version = None
+        alias = []
+        for tag in tags:
+            if re.match(r"\d+\.\d+\.\d+", tag):
+                version = tag
+            else:
+                alias.append(tag)
+
+        return cls(
+            uuid=img.uuid,
+            name=img.name,
+            title=img.name.split("/")[-1],
+            maintainer=img.maintainer,
+            released=img.released,
+            version=version,
+            version_aliases=alias,
+            **kwargs,
+        )
+
+
+class Job(BaseModel):
+    job_id: UUID
+    inputs_sha: str
+    solver_id: UUID
+
+    # hyperlinks
+    solver_url: HttpUrl
+    inspect_url: HttpUrl
+    outputs_url: HttpUrl
+
+
+# TODO: these need to be in sync with celery task states
+class TaskStates(str, Enum):
+    UNDEFINED = "undefined"
+    PENDING = "pending"
+    RUNNING = "running"
+    SUCCESS = "success"
+    FAILED = "failed"
+
+
+class JobStatus(BaseModel):
+    """
+
+    NOTE About naming. The result of an inspection on X returns a Status object
+        What is the status of X? What sort of state is X in?
+        SEE https://english.stackexchange.com/questions/12958/status-vs-state
+    """
+
+    job_id: UUID
+    state: TaskStates
+    progress: conint(ge=0, le=100) = 0
+
+    # Timestamps to some of the states
+    # TODO: sync state events and timestamps
+    submitted_at: datetime
+    started_at: Optional[datetime] = Field(
+        None,
+        description="Timestamp that indicate the moment the solver starts execution or None if the event did not occur",
+    )
+    stopped_at: Optional[datetime] = Field(
+        None,
+        description="Timestamp at which the solver finished or killed execution or None if the event did not occur",
+    )
+
+    def timestamp(self, event: str = "submitted"):
+        setattr(self, f"{event}_at", datetime.utcnow())
+
+
+class SolverPort(BaseModel):
+    name: KeyIdentifier = Field(
+        ...,
+        description="Name given to the input/output in solver specs (see solver metadata.yml)",
+    )
+
+    # TODO: define more specifically
+    #   - api/specs/common/schemas/node-meta-v0.0.1.json
+    #   - http://www.iana.org/assignments/media-types/media-types.xhtml
+    type: constr(
+        strict=True,
+        regex=r"^(number|integer|boolean|string|data:([^/\s,]+/[^/\s,]+|\[[^/\s,]+/[^/\s,]+(,[^/\s]+/[^/,\s]+)*\]))$",
+    ) = Field(None, description="Data type expected on this input/ouput")
+
+    title: Optional[str] = Field(
+        None, description="Short human readable name to identify input/output"
+    )
```

Comment:
these things are defined in the models-library. please use these. we do not want to have duplications.

Reply:
This is very important. Let's discuss it separately and block this PR until this is resolved.